src/gen_rules.ml

```diff
@@ -520,6 +522,49 @@ module Gen(P : Params) = struct
          >>>
          Build.and_create_file digest_path)
 
+  let copy_files_rules (def: Copy_files.t) ~src_dir ~dir ~package_context:_ =
+    let src_glob = SC.expand_vars sctx ~dir def.glob in
+    let glob_in_src = Path.relative src_dir src_glob in
```

Comment:
This doesn't enforce that the path must be a sub-directory of the current directory. You could write this instead:

```ocaml
Path.append src_dir (Path.relative Path.root src_glob)
```

and print a better error when `Path.relative Path.root src_glob` fails. Please capture the location of the glob as well so that the error is properly located.

BTW, there are many variables defined here and `src_glob` is used only once. I think it'd be easier to read if it was scoped accordingly:

```ocaml
let glob_in_src =
  let src_glob = ... in
  ...
in
```